cmdstanpy/model.py

```diff
@@ -894,6 +902,12 @@ def sample(
                 refresh = _CMDSTAN_REFRESH
             iter_total = iter_total // refresh + 2
 
+            if show_progress:
```

Comment:
I think we can simplify this to 
`show_progress = show_progress and _SHOW_PROGRESS`

Additionally, asserting types at runtime is usually frowned upon if it can be avoided. I think adding a type annotation in init.py can avoid this